stl/inc/atomic

```diff
@@ -2970,6 +2970,59 @@ inline void atomic_flag_notify_all(volatile atomic_flag* const _Flag) noexcept {
 inline void atomic_flag_notify_all(atomic_flag* const _Flag) noexcept {
     return _Flag->notify_all();
 }
+
+template <class _Ty>
+class _Locked_pointer {
+    static_assert(alignof(_Ty) >= (1 << 2), "2 low order bits are needed by _Locked_pointer");
+    static constexpr uintptr_t _Lock_mask                = 3;
+    static constexpr uintptr_t _Not_locked               = 0;
+    static constexpr uintptr_t _Locked_notify_not_needed = 1;
+    static constexpr uintptr_t _Locked_notify_needed     = 2;
+    static constexpr uintptr_t _Ptr_value_mask           = ~_Lock_mask;
+
+public:
+    [[nodiscard]] _Ty* _Lock_and_load() noexcept {

```

Comment:
Should that be `_NODISCARD` I am unsure what the convention on such internal helpers is

Reply:
We've been using `_NODISCARD` on internal machinery as well so the escape hatch for people who want to opt-out of our warnings also works to disable warnings if and when a user discovers an untested codepath in internal machinery which emits warnings. IIRC the only actual `[[nodiscard]]`s in the repo are in test code.

Reply:
We use `[[nodiscard]]` in `stl/src` (because it is separately compiled). We should probably think about removing the escape hatch, though - `[[nodiscard]]` appears to have been exceptionally well-tolerated due to our cautious policy (even though we aggressively emit these warnings in C++14 mode).

Reply:
This is a block guarded by `_HAS_CXX20` so I thought we didn't support an escape hatch there.

Reply:
I already changed `[[nodiscard]]` to `_NODISCARD` in barriers, do I need to change it back then?